rfcs/0006-gtfs-canonical-routes.md

```diff
@@ -0,0 +1,125 @@
+  shapes can have an identical set of points as those shapes used during current service. Existing
+  Commuter Rail shape references can be reused for canonical trips.
+- **trips.txt** includes one new trip for each canonical route/direction pair, or multiple trips in
+  the event of regular branching occuring along the route. It should reference the new `service_id`
+  created above. It should reference either a new (canonical) or previously-existing (regular
+  service) route pattern, depending on whether the trip's stopping pattern is currently served by
+  regular service. Each of these trips should be the `representative_trip_id` for their referenced
+  route patterns.
+- **stop_times.txt** includes all stops that trains in the route/direction passes through, even if
+  they are temporarily closed. Increment the arrival/departure times by one minute for each
+  subsequent stop. For a sample, see the attached GTFS file, the output of [✓ 🧪 🚝 Canonical
+  trips/patterns experiment](https://app.asana.com/0/881264583703207/1200210504369250), which
+  trialled this for Green Line D and Lowell Line (only in the outbound direction).
+
+This will be done as a new "step" of the Makefile called `canonical`, which runs between `truncate`
+and `current`. There will be a new a directory, `input/canonical/` to store any static data the step
+uses.
+
+The canonical step will be a Python module that reads in the `truncate` feed, and writes out its own
+feed after transforming it. The transformations are as above, and also will update any canonical
+trips to use current route patterns (and vice versa) if relevant, eliminating redundant route
+patterns.
+
+For Commuter Rail route patterns, the canonical one will specify every stop as non-flag, non-leave
+early stop. For both subway and Commuter Rail canonical trips, no data around bicycle allowances
+will be provided (`bikes_allowed=0`).
+
+# Drawbacks
+
+This is more complicated than not doing it.
+
+# Rationale and alternatives
+
+See proposed Concepts [1](https://github.com/mbta/technology-docs/pull/6#issuecomment-952315958), [4](https://github.com/mbta/technology-docs/pull/6#issuecomment-952315958), and [5](https://github.com/mbta/technology-docs/pull/6#issuecomment-962120434) in prior discussion.
+
+# Prior art
+
+Until now we've been doing ad-hoc adjustments to determine the route pattern of routes
+[here](https://github.com/mbta/api/blob/master/apps/state/config/config.exs#L145).
+
+# Unresolved questions
+
+Ensuring that we would not be in a situation where the sort order, perhaps ordered by
+frequency/proportion of trips, becomes a different order than desired for the main route pattern
+against the branch route pattern.
+
+How route patterns containing canonical representative trips are tagged or otherwise identified:
+Using a new `canonical_route_pattern` field in route_patterns.txt, or is the new
+`service_schedule_typicality` value sufficient?
```

Comment:
The current version of this RFC seems to resolve this question in favor of the former.

Reply:
It does, but I added the question myself to give folks an opportunity to question/challenge the resolution if they thought it was unnecessary.

Re your https://github.com/mbta/technology-docs/pull/6#discussion_r773282552, however, if we _don't_ make the "fake trips" representative of existing route patterns (when made canonical), then we would absolutely need a `canonical_route_pattern` field to show that canonical-ness.

Reply:
Doesn't the current approach require the field in either case? Even if the "fake trips" are always committed to the feed, you can't quite deduce the same information by looking up the service typicality of the trip (trip service typicality is 6 => pattern is canonical; trip service typicality is not 6 => ???, could be either that it is definitely-not-canonical or that no canonical patterns are defined for the route).

Reply:
You could probably infer between the two if you iterate over _all_ of the route patterns on a given route, but you're right in that it's a more tedious process. I'm more than happy to keep the `canonical_route_pattern` field, in any event.